be/src/exec/olap_scanner.cpp

```diff
@@ -474,6 +488,13 @@ Status OlapScanner::close(RuntimeState* state) {
     if (_is_closed) {
         return Status::OK();
     }
+    // olap scan node will call scanner.close() when finished
+    // will release resources here
+    // if not clear rowset readers in read_params here
+    // readers will be release when runtime state deconstructed but 
+    // deconstructor in reader references runtime state 
+    // so that it will core
```

Comment:
This comment make me confusing...

Reply:
It means, we should deconstructor resource in close method. Not in deconstructor method.